src/sentry/integrations/base.py

```diff
@@ -356,22 +356,18 @@ class IntegrationInstallation(abc.ABC):
     def __init__(self, model: RpcIntegration | Integration, organization_id: int) -> None:
         self.model = model
         self.organization_id = organization_id
-        self._org_integration: RpcOrganizationIntegration | None
 
-    @property
-    def org_integration(self) -> RpcOrganizationIntegration | None:
+    @cached_property
+    def org_integration(self) -> RpcOrganizationIntegration:
         from sentry.integrations.services.integration import integration_service
 
-        if not hasattr(self, "_org_integration"):
-            self._org_integration = integration_service.get_organization_integration(
-                integration_id=self.model.id,
-                organization_id=self.organization_id,
-            )
-        return self._org_integration
-
-    @org_integration.setter
-    def org_integration(self, org_integration: RpcOrganizationIntegration) -> None:
-        self._org_integration = org_integration
+        integration = integration_service.get_organization_integration(
+            integration_id=self.model.id,
+            organization_id=self.organization_id,
+        )
+        if integration is None:
+            raise NotFound("missing org_integration")
+        return integration
```

Comment:
this is the "big" change in this changeset that I'm not certain about -- it makes it so that we can't construct an integration with a missing `OrganizationIntegration`

I'm not certain that this is correct -- but it seems most of the integrations do not handle the case where this is missing and as far as I can understand it doesn't really make sense to have an integration not associated with an organization -- will need the domain owners to confirm this idea though

Reply:
I brought this up with my team to double check, and we think this is a fair assumption to make. Most of our integrations operate with the assumption that we have an org integration already, so it should be fine to handle the potentially few cases that don't explicitly with a try/catch.